Barotrauma/BarotraumaShared/Source/Characters/StatusEffect.cs

```diff
@@ -32,6 +32,8 @@ public enum TargetType
         public string[] propertyNames;
         private object[] propertyEffects;
 
+        List<Tuple<string, string, object>> propertyConditionals;
```

Comment:
I'd rather implement the conditionals as a separate class that handles the comparisons, operator parsing, etc. Would make it easier to reuse the conditional logic and List<PropertyConditional> would be less of an eyesore than List<Tuple<string, string, object>>. :)

Reply:
Yeah for sure. Ideally it should be like an <if target="Character" Health="< 50">, however that's problematic as it doesn't let you apply StatusEffect things like Health="50", etc.

Maybe turn it into its own class and add a new component that does exactly that? Since <if> component would be EXTREMELY useful for cases like this:
```xml
<if target="This" Condition="> 0.0">
  <StatusEffect="Always" Condition="-0.1"/>
</if>
<if target="This" Condition="<= 0.0">
  <StatusEffect="Always" .../>
</if>
```

Reply:
It shouldn't need to apply any StatusEffect things, the StatusEffect could simply call the relevant method in the PropertyConditionals to check if all the conditions are fullfilled before doing its StatusEffect business.

I'm not sure if there's a need for a separate component that only does those conditional checks - I think you could get the same result by simply putting the conditional and the StatusEffect inside any ItemComponent or use an empty base ItemComponent if there's no need for any additional ItemComponent functionality.

Reply:
I suppose so, yeah. I think my current implementation will need some testing to determine if it's robust enough.

Reply:
```
<if target="This" Condition="> 0.0">
  <StatusEffect="Always" Condition="-0.1"/>
</if>
```
I don't mean there's anything wrong with this approach, IMO it'd be a more clear way to define the conditionals and would take care of those "duplicate attribute" errors. I just meant the conditionals don't need to be a separate ItemComponent type, code-wise they could work just like they do now even if they were defined as separate XML elements.